### DISCUSSION TITLE
Handling OPENSSL_SYS_TANDEM API exposure in 3.2

### DISCUSSION THREAD

**Discussion by rsbeckerca** [openssl/openssl#23454] [General Discussion]
As of the 3.2 series, there is a significant `#if defined (OPENSSL_SYS_TANDEM)` in `include/openssl/e_os2.h` that must be set outside of OpenSSL for application builds, including curl. I am not sure how to best deal with this. Should this be an issue? A PR replacing `OPENSSL_SYS_TANDEM` with `__TANDEM`, or leave this alone. Does this have to happen in 3.3 or can it be fixed, if fixable, in 3.2?

https://github.com/openssl/openssl/blob/504e72fc1a1432d5266bd6e8909648c49884a36c/include/openssl/e_os2.h#L251C1-L253C24

I also think this is a broader impact issue that might hit other platforms, with `e_os2.h` being used in the `#include` path.

**Comment by t8m** [comment]
I think replacing with `__TANDEM` is the way to go and it looks like a bug fix to me -> applicable to 3.2 as well.

**Comment by t8m** [comment]
Hmm, I am actually taking my words back. Why should this be an issue?? The e_os2.h file includes openssl/opensslconf.h which in turn includes openssl/configuration.h which is supposed to define this macro on NonStop.

**Comment by rsbeckerca** [comment]
> Hmm, I am actually taking my words back. Why should this be an issue?? The e_os2.h file includes openssl/opensslconf.h which in turn includes openssl/configuration.h which is supposed to define this macro on NonStop.

I think you are correct here. The problem is that the path through the source is missing `#include <stdint.h>`, which is causing the **curl** 8.6.0 build to fail at 3.2 but not 3.0. The only relevant change I found so far was in e_os2.h but that obviously isn't it.

**Comment by rsbeckerca** [comment]
> Hmm, I am actually taking my words back. Why should this be an issue?? The e_os2.h file includes openssl/opensslconf.h which in turn includes openssl/configuration.h which is supposed to define this macro on NonStop.

I think this comes down to operator error. I forgot a definition for curl that I had to add to the OpenSSL build some months ago. My bad. Thanks for your input, @t8m
